### PR TITLE
Added ability to change planner configurations in the interface

### PR DESCRIPTION
**Pull request by davetcoleman**
This is for parameter sweeping

### CODE REVIEW

File: planning_interface/include/moveit/planning_interface/planning_interface.h [hunk 1]
```diff
@@ -45,6 +45,15 @@
 namespace planning_interface
 {
 
+/// Holds settings specific for each planner, as found in i.e. ompl_planning.yaml
```
Comment (isucan) [inline planning_interface/include/moveit/planning_interface/planning_interface.h]: This file cannot be specific to ompl; We should remove the reference above

File: planning_interface/include/moveit/planning_interface/planning_interface.h [hunk 2]
```diff
@@ -45,6 +45,15 @@
...
+struct PlanningConfigurationSettings
+{
+  std::string                        name;
```
Comment (isucan) [inline planning_interface/include/moveit/planning_interface/planning_interface.h:55]: We should document things here; say how the settings are used. 
I suggest the following policy: settings with unknown keys are ignored.
Settings for unknown groups are ignored.

File: planning_interface/include/moveit/planning_interface/planning_interface.h [hunk 3]
```diff
@@ -78,6 +87,12 @@ class Planner
   
   /// Determine whether this plugin instance is able to represent this planning request
   virtual bool canServiceRequest(const MotionPlanRequest &req)  const = 0;
+
+  /// Settings for planners to use for their specific algorithms
+  virtual void setPlanningConfigurations(const PlanningConfigurationMap &pcs) const = 0;
+
+  /// Get settings for planners to use for their specific algorithms, so that they can then be modified
+  virtual const PlanningConfigurationMap getPlanningConfigurations() const = 0;
```
Comment (isucan) [inline planning_interface/include/moveit/planning_interface/planning_interface.h:101]: should we return by value here? Or should we fill an arg? Maybe this is fine as it is.

File: planning_interface/include/moveit/planning_interface/planning_interface.h (hunk 2 above)
Comment (davetcoleman) [inline planning_interface/include/moveit/planning_interface/planning_interface.h:55]: I don't completely remember how this struct works - it came from OMPL and I just added it to MoveIt. I have documented what I can.

File: planning_interface/include/moveit/planning_interface/planning_interface.h (hunk 3 above)
Comment (davetcoleman) [inline planning_interface/include/moveit/planning_interface/planning_interface.h:101]: no opinion.

**Pull request merged**